### DISCUSSION TITLE
How would I consider porting the i2s and SPI to native Arduino SPI.h

### DISCUSSION THREAD

**Discussion by 8888clockradio** [micropython/micropython#9833] [Core Development]
How would I consider porting the i2s and SPI to native Arduino SPI.h in the mimixrt port.
I need to port both of these to native Arduino C++, how would I go about updating them. I get I have to make it a C++ and use #ifdef __cplusplus
extern "C" {
#endif 

#ifdef __cplusplus
}
#endif

**Comment by jimmo** [comment]
Not quite sure what you're asking sorry. The mimxrt port doesn't use anything from the Arduino SDK.

Can you give an example of what you're trying to do?
